lib/spack/spack/test/llnl/util/tty/log.py

```diff
@@ -62,6 +63,23 @@ def test_log_python_output_without_echo(capfd, tmpdir):
         assert capfd.readouterr()[0] == ''
 
 
+def test_log_python_output_with_invalid_utf8(capfd, tmpdir):
+    with tmpdir.as_cwd():
+        with log_output('foo.txt'):
+            sys.stdout.buffer.write(b'\xc3\x28\n')
+
+        if sys.version_info.major == 2:
+            with open('foo.txt') as f:
+                assert(b'\xc3(\n')
```

Comment:
```suggestion
            with open('foo.txt', 'rb') as f:
                assert(b'\xc3(\n') in f.read()
```

I think you forgot to read here?

Reply:
Thanks, applied.